catalyst_voices_packages/catalyst_key_derivation/rust/src/api/key_derivation/mod.rs

```diff
@@ -0,0 +1,116 @@
+//! Key derivation module.
+
+use bip32::DerivationPath;
+use bip39::Mnemonic;
+pub use ed25519_bip32::DerivationIndex;
+pub use ed25519_bip32::DerivationScheme;
+pub use ed25519_bip32::XPrv;
+use hmac::Hmac;
+use pbkdf2::pbkdf2;
+use sha2::Sha512;
+
+/// Extended private key bytes type
+pub type XPrvBytes = [u8; 96];
+
+/// Generate a new extended private key (`XPrv`) from a mnemonic and passphrase.
+/// This function works with BIP-0039 mnemonics.
+/// For more information: Cardano Icarus master node derivation
+/// https://github.com/satoshilabs/slips/blob/master/slip-0023.md
+/// 
+/// # Arguments
+///
+/// - `mnemonic`: A string representing the mnemonic.
+/// - `passphrase`: An optional string representing the passphrase.
+///
+/// # Returns
+///
+/// Returns the `XPrv` extended private key as a `Result`.
+/// If the conversion is successful, it returns `Ok` with the extended private key
+/// (`XPrv`).
+pub fn mnemonic_to_xprv(mnemonic: String, passphrase: Option<String>) -> anyhow::Result<XPrvBytes> {
+    let xprv = mnemonic_to_xprv_helper(mnemonic, passphrase)?;
```

Comment:
Make sure to run it in a thread_pool [like here](https://github.com/input-output-hk/catalyst-voices/blob/main/catalyst_voices_packages/catalyst_key_derivation/rust/src/api/simple.rs#L12) to avoid blocking the main thread on Flutter side